project/app/views.py

```diff
@@ -11,31 +11,32 @@
 @api_view(['GET'])
 def apiOverview(request):
     api_urls = {
-        'List': '/task-list/',
-        'Detail View': '/task-detail/<str:pk>/',
-        'Create': '/task-create/',
-        'Update': '/task-update/<str:pk>/',
-        'Delete': '/task-delete/<str:pk>/',
+        'sign-up': '/sign-up/',
+        'login': '/login/',
     }
     return Response(api_urls)
 
 
-@api_view(['GET'])
-def taskList(request):
-    tasks = Task.objects.all()
-    serializer = TaskSerializer(tasks, many=True)
-    return Response(serializer.data)
-
-
-@api_view(['GET'])
-def viewone(request, pk):
-    tasks = Task.objects.get(id=pk)
-    serializer = TaskSerializer(tasks,many=False)
+@api_view(['POST'])
+def signUp(request):
+    serializer = SignUpSerializer(data=request.data)
+    if serializer.is_valid():
+        serializer.save()
     return Response(serializer.data)
 
 
-@api_view(['POST'])
-def update(request,pk):
-    tasks = Task.objects.get(id=pk)
-    serializer = TaskSerializer(tasks, data=request.data)
-    return Response(serializer.data)
\ No newline at end of file
+@api_view(["POST"])
+def login(request):
+    serializer = LoginSerializer(data=request.data)
+    if serializer.is_valid():
+        user = serializer.data["username"]
+        pwd = serializer.data["password"]
+        try:
+            details = SignUp.objects.get(username=user, password=pwd)
```

Comment:
Raw password string we are saving in model that is not correct.
We should save hash password and then we can check the password using `authenticate` method or using default `check_password` method.